internal/core/entities/pdb/pdb.go

```diff
@@ -0,0 +1,81 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package pdb
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/topfreegames/maestro/internal/core/ports/errors"
+	policyv1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Pdb MaxUnavailable specifies the budget to limit evictions of pods inside a namespace
+type Spec = policyv1.PodDisruptionBudgetSpec
+
+// Type represents the stored type of IntOrString.
+type Type = intstr.Type
+
+const (
+	Int    Type = iota // The IntOrString holds an int.
+	String             // The IntOrString holds a string.
+)
+
+func ValidateDefault(defaultVal string) (string, error) {
+	if strings.HasSuffix(defaultVal, "%") {
+		percentageStr := strings.TrimSuffix(defaultVal, "%")
+		percentage, err := strconv.Atoi(percentageStr)
+		if err != nil || percentage < 0 || percentage > 100 {
```

Comment:
if we dont allow `< 1` if not percentage, why allow here?

Reply:
Good point. 1% might be a valid percentage, but probably allow setting 1 as non-percentage number as well. WDYT?

Reply:
I've removed this function in favor of the API one that specifies >0 and <100 for percentage string and >0 for raw numbers

Reply:
great!